bciers/apps/registration/tests/components/form/widgets/MultiSelectWidget.test.tsx

```diff
@@ -0,0 +1,183 @@
+      },
+    },
+  },
+} as RJSFSchema;
+
+export const multiSelectFieldUiSchema = {
+  multiSelectTestField: {
+    "ui:widget": "MultiSelectWidget",
+  },
+};
+
+describe("RJSF MultiSelectWidget", () => {
+  it("should render a combo box field", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+    expect(screen.getByLabelText(multiSelectLabelRequired)).toBeVisible();
+  });
+
+  it("should render the combo box value when formData is provided", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        formData={{ multiSelectTestField: ["option_2"] }}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    // Each displayed value is a MUI Chip/button
+    expect(screen.getByRole("button", { name: "Option 2" })).toBeVisible();
+  });
+
+  it("should render multiple combo box values when formData is provided", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        formData={{ multiSelectTestField: ["option_1", "option_3"] }}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    expect(screen.getByText("Option 1")).toBeVisible();
+    expect(screen.getByText("Option 3")).toBeVisible();
+  });
+
+  it("should allow selecting a combo box value", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const openMultiSelectButton = screen.getByRole("button", { name: "Open" });
+    await userEvent.click(openMultiSelectButton);
+
+    const option3 = screen.getByText("Option 3");
+
+    await userEvent.click(option3);
+
+    expect(screen.getByRole("button", { name: "Option 3" })).toBeVisible();
+  });
+
+  it("should allow typing a combo box value", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const multiSelectInput = screen.getByRole("combobox") as HTMLInputElement;
+
+    await userEvent.type(multiSelectInput, "Option 3{enter}");
+
+    expect(screen.getByRole("button", { name: "Option 3" })).toBeVisible();
+  });
+
+  it("should allow clearing the combo box", async () => {
+    render(
+      <FormBase
+        schema={multiSelectFieldSchema}
+        formData={{ multiSelectTestField: ["option_2", "option_3"] }}
+        uiSchema={multiSelectFieldUiSchema}
+      />,
+    );
+
+    const selectedOption1 = screen.getByRole("button", { name: "Option 2" });
+    const selectedOption2 = screen.getByRole("button", { name: "Option 3" });
+    const optionCancelIcon = screen.getByTestId("CloseIcon");
+
+    await userEvent.click(optionCancelIcon);
+
+    expect(selectedOption1).not.toBeInTheDocument();
+    expect(selectedOption2).not.toBeInTheDocument();
+  });
+
+  // TODO: This is currently broken in MultiSelectWidget
```

Comment:
Should these commented tests stay here as a reminder or go?

Reply:
We should have a ticket for any TODOs (or Dylan will be sad), but I don't think anyone has strong opinions on whether to leave comments in the code. I personally like them because they make it easy to find where the problem is

Reply:
@BCerki there is a link to the issue in below the TODO comment, sorry I should have made a comment here like the other one https://github.com/bcgov/cas-registration/issues/1602